File: Dependant variables_blazor-server/DependantVariables/Services/NorthWindv2APIService.cs
using System.Net.Http.Json;
using System.Reactive.Subjects;
using DependantVariables.Models.NorthWindv2API;

namespace DependantVariables.NorthWindv2API
{
    public class NorthWindv2APIService: INorthWindv2APIService
    {
        private readonly HttpClient _http;

        public NorthWindv2APIService(HttpClient http)
        {
            _http = http;
        }

        public BehaviorSubject<CustomerDto> SelectedCustomer { get; set; } = new(default);
        public BehaviorSubject<OrderDto> SelectedOrder { get; set; } = new(default);

        public async Task<List<CustomerDto>> GetCustomerDtoList()
        {
            using HttpRequestMessage request = new HttpRequestMessage(HttpMethod.Get, new Uri("https://data-northwind.indigo.design/Customers", UriKind.RelativeOrAbsolute));
            using HttpResponseMessage response = await _http.SendAsync(request).ConfigureAwait(false);
            if (response.IsSuccessStatusCode)
            {
                return await response.Content.ReadFromJsonAsync<List<CustomerDto>>().ConfigureAwait(false);
            }

            return new List<CustomerDto>();
        }

        public async Task<List<OrderDetailDto>> GetOrderDetailDtoList(string id)
        {
            using HttpRequestMessage request = new HttpRequestMessage(HttpMethod.Get, new Uri($"https://data-northwind.indigo.design/Orders/{id}/Details", UriKind.RelativeOrAbsolute));
            using HttpResponseMessage response = await _http.SendAsync(request).ConfigureAwait(false);
            if (response.IsSuccessStatusCode)
            {
                return await response.Content.ReadFromJsonAsync<List<OrderDetailDto>>().ConfigureAwait(false);
            }

            return new List<OrderDetailDto>();
        }

        public async Task<List<OrderDto>> GetOrderDtoList(string id)
        {
            using HttpRequestMessage request = new HttpRequestMessage(HttpMethod.Get, new Uri($"https://data-northwind.indigo.design/Customers/{id}/Orders", UriKind.RelativeOrAbsolute));
            using HttpResponseMessage response = await _http.SendAsync(request).ConfigureAwait(false);
            if (response.IsSuccessStatusCode)
            {
                return await response.Content.ReadFromJsonAsync<List<OrderDto>>().ConfigureAwait(false);
            }

            return new List<OrderDto>();
        }
    }
}

File: Dependant variables_angular/src/app/services/north-windv2-api.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { BehaviorSubject, Observable } from 'rxjs';
import { CustomerDto } from '../models/north-windv-2api/customer-dto';
import { OrderDetailDto } from '../models/north-windv-2api/order-detail-dto';
import { OrderDto } from '../models/north-windv-2api/order-dto';

const API_ENDPOINT = 'https://data-northwind.indigo.design';

@Injectable({
  providedIn: 'root'
})
export class NorthWindv2APIService {
  constructor(
    private http: HttpClient
  ) { }
  
  public selectedOrder: BehaviorSubject<OrderDto | undefined> = new BehaviorSubject<OrderDto | undefined>(undefined);

  private _selectedCustomer$!: BehaviorSubject<CustomerDto | undefined>;

  public get selectedCustomer(): BehaviorSubject<CustomerDto | undefined> {
    if (!this._selectedCustomer$) {
      this._selectedCustomer$ = new BehaviorSubject<CustomerDto | undefined>(undefined);
      this._selectedCustomer$.subscribe(() => this.selectedOrder.next(undefined));
    }

    return this._selectedCustomer$;
  }


  public getCustomerDtoList(): Observable<CustomerDto[]> {
    return this.http.get<CustomerDto[]>(`${API_ENDPOINT}/Customers`);
  }

  public getOrderDetailDtoList(id: string): Observable<OrderDetailDto[]> {
    return this.http.get<OrderDetailDto[]>(`${API_ENDPOINT}/Orders/${id}/Details`);
  }

  public getOrderDtoList(id: string): Observable<OrderDto[]> {
    return this.http.get<OrderDto[]>(`${API_ENDPOINT}/Customers/${id}/Orders`);
  }
}
